.gitmodules

```diff
@@ -22,3 +22,6 @@
 [submodule "external/libpng"]
   path = external/libpng
   url = https://github.com/glennrp/libpng.git
+[submodule "external/llvm-project"]
```

Comment:
Phew, this looks like a veeery fat dependency with a lot of unnecessary stuff that gets downloaded, too.  According to [this StackExchange entry](https://stackoverflow.com/questions/600079/how-do-i-clone-a-subdirectory-only-of-a-git-repository), github supports `git clone --filter`, so I wonder whether this can be streamlined.

Reply:
Will the submodule stuff for LLVM happen unconditionally or only if asked for MemorySanitizer support in `custom-build.sh`?

Reply:
Here I'm always building it. I had considered making this a MemorySanitzer only thing, but this ended up being more complex, not saving any real time, and being less reproducible. In oss-fuzz everything is supposed to be statically linked anyway. Dynamically linking against libc++ or libstdc++ as was being done before was actually a bit iffy (though I'm sure there are many projects in oss-fuzz implicitly depending on this). Also, by building everything with the sanitizer everything gets checked, so we won't have issues like the libpng issue where one library calculates bad values but another uninstrumented library uses it and the issue isn't detected.

That being said, building libcxx and libcxx-abi doesn't take long (it's only ~550 build steps and locally doesn't take that long). Cloning llvm-project does take a bit as it is large at 4GB or so for a full clone, though I haven't looked into how big a shallow clone is. I don't think there's any good way to filter a submodule, though this might work with subtree. Even if we could get just part of llvm-project doesn't appear to really ensure that partial checkouts will always work fro building.

Reply:
Now that we can rely on `--depth 1` for submodules, git reports that it only pulls down a total of 22.00 MiB for llvm-project. Which isn't nearly as big or take nearly as long as I thought it would. So this isn't quite as fat as one might initially think. Admittedly, it isn't tiny either and is probably still the largest thing pulled by this build.